### ISSUE TITLE
ip_helper_generic should ignore non running interfaces

### ISSUE THREAD

**Issue by AmarOk1412** [pjsip/pjproject#2686]
**Describe the bug**

During ICE negotiations ips candidates from non running interfaces (generally from virtual machines down or docker) are added for example:

```
8: docker0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue
state DOWN group default
     link/ether 02:42:a1:ff:d0:4d brd ff:ff:ff:ff:ff:ff
     inet 172.17.0.1/16 brd 172.17.255.255 scope global docker0
        valid_lft forever preferred_lft forever
     inet6 fe80::42:a1ff:feff:d04d/64 scope link
        valid_lft forever preferred_lft forever
```

Even if state=DOWN the candidate 172.17.0.1 will be added.

I think this is not wanted. In pjlib/src/pj/ip_helper_generic.c the ip
should not be added if interface is down (and a check is there for that
(it->ifa_flags & IFF_UP). However, this check doesn't seems to be enough
and I think IFF_RUNNING is the flag intended here
(https://man7.org/linux/man-pages/man7/netdevice.7.html)
 > IFF_UP Interface is running.

 > IFF_RUNNING Resources allocated.

Because I think it should get rid of "NO-CARRIER,UP" links like my docker0

**To Reproduce**

1. Start & stop a docker container, this will typically add `docker0 (...) state DOWN` when typing `ip a`. (Or start/stop a virtual machine)
2. Launch a ICE negotiation

**Expected behavior**
Because the interface is not running, ip should not be added

**Desktop/Smartphone (please complete the following information):**

- OS, Distribution & Version: GNU/Linux (Android & macOS should be affected imho)
- PJSIP
  - version: 2.10, but 2.11 is the same
  - applied patch(es): https://git.jami.net/savoirfairelinux/ring-daemon/blob/master/contrib/src/pjproject/rules.mak describes how it's build

**Comment by mchibani-sfl** [comment]
+1

**Comment by sauwming** [comment]
Fixed in #2687
